lib/src/food.ts

```diff
@@ -0,0 +1,205 @@
+import { ThresholdTable } from './rollFromTable'
+
+type CookingMethods =
+| ''
+| 'ingredient'
+| 'raw'
+| 'boiled'
+| 'sauteed'
+| 'roasted'
+| 'pickled'
+| 'braised'
+| 'toasted'
+| 'fried'
+| 'fermented'
+| 'spiced'
+
+type DishTypes =
+  'main'
+| 'entree'
+| 'dessert'
+| 'side'
+
+type IngredientNames = keyof typeof ingredients
+
+/** The 'dairy', 'meat', 'seafood' and 'animal product' autoflag their non-vegan / vegetarian-ness. */
+type IngredientCategories =
+  'staple'
+| 'grain'
+| 'meat'
+| 'dairy'
+| 'seafood'
+| 'animal product'
+| 'fruit'
+| 'vegetable'
+| 'berry'
+| 'spice'
+| 'baking'
+| 'alcohol'
+
+type CookingElements =
+  'sauce'
+| 'crust'
+| 'garnish'
+| ''
+
+type DishTastes =
+  'savoury'
+| 'sweet'
+
+interface Dishes {
+  /** Override, for when you want a fancy name for your dish. */
+  name?: string
+  /** Override, for when there are multiple variants (chicken curry vs chickpea curry)
+   * The sub-dish needs to specify ingredients that are specific to it- common ingredients go in `ingredients`.
+  */
+  types?: Dishes[]
+  /** Flagging `isVegan` will automatically flag `isVegetarian` as true. */
+  isVegetarian?: boolean
+  /** Flagging `isVegan` will automatically flag `isVegetarian` as true. */
+  isVegan?: boolean
```

Comment:
These could probably be combines into a single attribute. Perhaps something along the lines of  `diet: "vegan"`.

Reply:
I was thinking that the simplest way to calculate veganocity would be an optional override, and then a simple if statement to check whether there were any `ingredients` that broke the conduct.

Reply:
Hm, seems it might break the single.soutce of truth principle. Perhaps we could use an `isVegan`, and `isVegetarian` function to check it at runtime instead?

Reply:
Yep, that's the plan.

Reply:
And then we don't need the flags to be on the object, as they're calculated instead of explicitly set.